src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs

```diff
@@ -3177,12 +3177,19 @@ public void GetContainerProperties()
         /// <summary>
         /// Dynamically load the Host Compute interop assemblies and return useful types.
         /// </summary>
-        /// <param name="computeSystemPropertiesType">The Microsoft.HyperV.Schema.Compute.System.Properties type.</param>
+        /// <param name="computeSystemPropertiesType">The HCS.Compute.System.Properties type.</param>
         /// <param name="hostComputeInteropType">The Microsoft.HostCompute.Interop.HostComputeInterop type.</param>
         private static void GetHostComputeInteropTypes(out Type computeSystemPropertiesType, out Type hostComputeInteropType)
         {
             Assembly schemaAssembly = Assembly.Load(new AssemblyName("Microsoft.HyperV.Schema, Version=10.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35"));
-            computeSystemPropertiesType = schemaAssembly.GetType("Microsoft.HyperV.Schema.Compute.System.Properties");
+
+            // The type name was changed in newer version of Windows so we check for new one first,
+            // then fallback to previous type name to support older versions of Windows
+            computeSystemPropertiesType = schemaAssembly.GetType("HCS.Compute.System.Properties");
```

Comment:
Is the both assembly APIs public? What about compliance?

Reply:
That's an excellent question.  I'll follow-up on that.

Reply:
The apis being used are published at https://github.com/microsoft/dotnet-computevirtualization which is considered sufficient for compliance

Reply:
I see another dll vmcompute.dll is used there not Microsoft.HyperV.Schema and Microsoft.HostCompute.Interop

Reply:
I see the same pinvokes in the repo.  I'll follow-up again to see if those exported functions are documented as i couldn't find them on docs.microsoft.com

Reply:
The response from Hyper-V team is that the .Net wrapper serves as documentation so from a compliance perspective, we are good here.  They also inform me that documentation on the pinvoked'd win32 api's should be documented in a few weeks.

Reply:
So we should migrate to vmcompute.dl?